plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/RequestMetricCollector.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.repositories.azure;
+
+/**
+ * A listener that can be hooked into any Azure Storage request allowing metric collection
+ */
+interface RequestMetricCollector {
+
+    /**
+     * Used to collect metrics after a successful request/response cycle
+     *
+     * @param requestMethod HTTP request method
+     */
+    void collectMetrics(String requestMethod);
```

Comment:
Since we didn't actually implement the failed request counting (for now at least), maybe just use a `Consumer<String>` where this is used and simplify it away for now (I think we can just collect the failures to some string key as well if we want to do that going forward)?

Reply:
Yes, let's simplify and get rid of this 👍

Reply:
For POST and PUT operations we would need a `BiConsumer` as the distinction lies in the URL query params.

Reply:
That or just make it a consumer of `HttpURLConnection` (might be a little easier to read and the duplication is trivial IMO)?

Reply:
👍